DocCRender.docc/contributing/Internals.md

```diff
@@ -0,0 +1,189 @@
+
+Learn how DocC Render is built internally, the principles behind its structure and how the components interact.
+
+## Overview
+
+DocC Render supports 4 different page types out of the box, with ability for adding custom pages. These pages create a well-structured documentation for your project, consisting of reference pages, tutorials and articles.
+
+Each page fetches its required render JSON asynchronously, from the `/data` directory. To learn more about the types of pages DocC Render supports, visit the [DocC Docs](https://www.swift.org/documentation/docc).
+
+### Modular Architecture
+
+The `swift-docc-render` package holds a variety of components, useful to build pages. All components are packaged as raw Vue files, so your App's bundler can take care of transpilation and packing them.
+
+### Vue Page Structure
+
+Routes in DocC Render are rendered using VueRouter. Pages can be dynamic, meaning a single page can match many urls, but still be rendered by the same Vue Page Component. For example the `/tutorial/:slug` route is rendered by the **TutorialOverview** page.
+
+For each page type we have:
+
+* A **View** component inside the `views` folder. It is used to fetch and provide the data down to its children.
+* A **Page** component, responsible for rendering the Atomic components, based on the available JSON. This is generally a direct child component of the **View** component.
+
+### Sub Components
+
+Each page is made out of isolated sub components, rendered depending on the data coming from DocC. These components are the building blocks of all pages.
+
+Some of the more generic ones like `ContentNode` or `GridColumn` are in almost every page, whereas others belong to only certain page types. Those belonging to a certain page type, are usually scoped to a folder named after their page type.
+
+### Internal Links and Routes
+
+Routes in DocC Render are very flexible and define a very relaxed matching pattern.
+
+For example the root `/documentation/` path for the Docs pages is hardcoded, but for anything nested after it, DocC Render will try to fetch data and render a page. Same goes for the `/tutorials/` page type.
+
+**Example:** 
+For the route `/documentation/path/to/something` DocC Render will try to fetch DocC generated JSON, using the same nested path `/data/documentation/path/to/something.json`. 
+
+Links to pages in DocC Render are generated using references coming from the render JSON. You can use `ReferenceUrlProvider`, to get the URL and title from a reference.
+
+```html
+<ReferenceUrlProvider :reference="someFooReference">
+    <router-link 
+        slot-scope="{ url, title }"
+        :to="url"
+    >
+        {{ title }}
+    </router-link>
+</ReferenceUrlProvider>
+```
+
+### Stores
+
+A DocC Render store module extracts commonly used data, that needs to be reactive and be accessible to multiple components. A store is just an ES Module, that exposes a `state` object and some methods.
```

Comment:
Maybe we could add a small note explaining that this is basically just the approach described in the [Simple State Management from Scratch](https://vuejs.org/v2/guide/state-management.html#Simple-State-Management-from-Scratch) section of the Vue docs.